### PR TITLE
updated main page layout and stylings

### PR DESCRIPTION
**Pull request by Lada496**
### Feature Description
1. added stylings for main page along with @SachiGoto's design [Figma](https://www.figma.com/file/5JTHosopzXXjY5BFxp8vMf/Map-Project?type=design&node-id=0-1&t=yNYjQVDQOkfUdozr-0)
2. added global layout according to [our vote](https://devjoylee.notion.site/devjoylee/Mantine-UI-70c0a9739a9e4fb2bdb34aa04a7b9f4a). There are the code references for each UI: [Header,](https://ui.mantine.dev/component/header-responsive) [Footer](https://ui.mantine.dev/component/footer-social), [ToggleSwitch](https://ui.mantine.dev/component/switch-toggle) 
3. set up dark mode (ref: [set up for tailwind](https://tailwindcss.com/docs/dark-mode), [set up for manine](https://mantine.dev/guides/dark-theme/))
4. set up custom colors for tailwind (**temporal**)

### Related Issue
N/A

### Something you want to mention

- We need to replace the temporal buttons form @SachiGoto's buttons later
- I think dark mode colors need to be customized later or @devjoylee's update??
- I realized we also need custom colors for tailwind and just added them. Please let me know if you have any naming preference.
### Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/clients/components/ThemeSwitch.tsx
Comment (devjoylee) [inline src/clients/components/ThemeSwitch.tsx:10]:
```suggestion
const ThemeSwitch = () => {
  // eslint-disable-next-line @typescript-eslint/unbound-method
```
image

I had this same `UNBOUND-METHOD` error by eslint. (You will find the error on Netlify deploy log)
So I added this comment line and the error was fixed!

File: tailwind.config.ts
Comment (devjoylee) [inline tailwind.config.ts:1]: I also created a theme object but using Mantine. 
We should decide what to use for our theme object I guess..

```js
import type { MantineThemeOverride } from "@mantine/core";

const theme: MantineThemeOverride = {
  colors: {
    blue: ["#e0fbfc", "#98c1d9", "#3d5a80", "#293241"],
    red: ["#ee6c4d"],
  },
  primaryColor: "blue",
  primaryShade: 2,
};

export default theme;
```

**Review by devjoylee — COMMENTED** [review]
I am afraid we have some duplicated codes because I also created a theme provider and a theme toggle button this time.  But don't worry , I used same reference as you mentioned in description  so most of them look similar. I can just combine them and delete duplicated codes after I merged yours!

File: tailwind.config.ts
Comment (Lada496) [reply tailwind.config.ts:1]: I can use "blue" and "red" instead of the current names :)
I haven't checked yet but I guess taildwind can't use Mantine's color so we actually need both.
But let me double check

File: src/clients/components/ThemeSwitch.tsx
Comment (Lada496) [reply src/clients/components/ThemeSwitch.tsx:10]: > I am afraid we have some duplicated codes because I also created a theme provider and a theme toggle button this time. But don't worry , I used same reference as you mentioned in description so most of them look similar. I can just combine them and delete duplicated codes after I merged yours!

Oh apologies for not having enough conversation about that.
I just included all code related to Header and Footer basically...

Comment (devjoylee) [reply src/clients/components/ThemeSwitch.tsx:10]: No worries. It's all fine!
I should've noticed the toggle button belongs to Header as well.

**Review by devjoylee — APPROVED** [review]

**Review by SachiGoto — APPROVED** [review]

**Pull request merged**
